### ISSUE TITLE
[Bug]: MessageBar doesn't announce message with SRs

### ISSUE THREAD

**Issue by bansalrachita** [microsoft/fluentui#25311]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: macOS 12.6
  Browsers:
    Chrome: 106.0.5249.119
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/bansalrachita/pen/qBYzazP

## Actual Behavior
MessageBar doesn't announce the message when messageBarType is set to info/success.
SR combinations used: Web-Chrome + NVDA and Android + Talkback

## Expected behavior: 
Should announce the message without setting role to "alert". 

## Workaround: 
Works fine if role is set to "alert".

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
